bootstrap_cfn/fab_tasks.py

```diff
@@ -148,16 +169,111 @@ def user(username):
     env.user = username
 
 
+def generate_stack_name():
+    """
+    Used to generate new stack name.
+
+    Format <application>-<environment>-<tag>
+
+    tag is optional and defaults to 'active'
+    """
+    if hasattr(env, 'tag'):
+        tag = env.tag
+    else:
+        tag = 'active'
+        env.tag = tag
+    legacy_name = "{0}-{1}".format(env.application, env.environment)
+    cfn_config = get_config()
+    try:
+        r53_conn = get_connection(R53)
+        zone_name = cfn_config.data['master_zone']
```

Comment:
Looks like I've covered that by catching the KeyError and returning the legacy_name?